### PR TITLE
FIX: Convert LTA to ITK with nitransforms

### LINKED ISSUE #185: Bringing T1w image into template space
Hello.
I have a multi-session study and am trying to bring a T1w image into template space using `antsApplyTransforms` with 
`sub-{subject}/ses-{session}/anat/*_T1w_space-orig_target-T1w_affine.txt`

However, the resulting image is completely flipped around (orientation labels are off) and cropped. The T1w is LAS oriented, but bringing it to RAS first didn't do anything?

Do you have any advise how to proceed? Thanks.

### PR DESCRIPTION
**Pull request by effigies**
Closes #185.

### CODE REVIEW

File: smriprep/interfaces/nitransforms.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    if reference is not None and not xfm.reference:
+        xfm.reference = reference
+
+    xfm.to_filename(out_file, moving=moving, fmt=fmt)
+
+    if out_inv is not None:
+        inv_xfm = ~xfm
+        if moving is not None:
+            inv_xfm.reference = moving
+        (~xfm).to_filename(out_inv, moving=reference, fmt=fmt)
```
Comment (oesteban) [inline smriprep/interfaces/nitransforms.py:81]: I don't know why I did this, but it looks like I really wanted to do this:
```suggestion
        inv_xfm.to_filename(out_inv, moving=reference, fmt=fmt)
```

I spotted this in fMRIPrep but failed to report. Leaving the note here for future consideration.

**Comment by oesteban** [comment]
BTW, would this (=nitransforms integration) be part of 20.1?

**Comment by effigies** [comment]
Hmm. Was that not already party of fmriprep?

**Comment by oesteban** [comment]
Correct.

I integrated nitransforms for that problem because the alternative seemed even finickier (nipype conditionals and more third-party tooling). Nitransforms allowed me to read LTA (when running the full pipeline) and ITK (when using the anatomical fast-track) very easily.

I guess we can say the experimenting with nitransforms is done, and proceed with a more global (or _full_ using my own term) integration.

I'm happy with that, just making sure we are all on the same page 👍

**Comment by effigies** [comment]
For 20.1, I'd say let's transition only things that need adjustment for some reason. We can make a full transition an explicit goal of 20.2.

**Comment by oesteban** [comment]
Okay, how that approach materializes in practice? I mean, you are proposing:

a) having fmriprep's and smriprep's copies of this code coliving until 20.2; or
b) upstreaming the code to niworkflows (so there's no duplicity), but refrain from including more features of nitransforms beyond this interface, until 20.2

**Comment by effigies** [comment]
b, more or less.

I just meant "let's not transition convert3d nodes that are working". If it seems worth it to upstream the interface to niworkflows because it's now in two places, that's okay.

**Comment by oesteban** [comment]
sounds great. I'll send a PR to niworkflows moving the code, tomorrow.

File: smriprep/workflows/anatomical.py [hunk 2]
```diff
@@ -614,12 +612,12 @@ def _set_threads(in_list, maximum):
...
-        (t1w_conform_xfm, concat_affines, [('out_lta', 'in_lta1')]),
-        (t1w_merge, concat_affines, [('transform_outputs', 'in_lta2')]),
-        (concat_affines, lta_to_itk, [('out_file', 'in_lta')]),
+        (t1w_conform_xfm, merge_xfm, [('out_lta', 'in1')]),
+        (t1w_merge, merge_xfm, [('transform_outputs', 'in2')]),
+        (merge_xfm, concat_xfms, [('out', 'in_xfms')]),
         # Output
         (t1w_reorient, outputnode, [('out_file', 't1w_ref')]),
-        (lta_to_itk, outputnode, [('out_itk', 't1w_realign_xfm')]),
+        (concat_xfms, outputnode, [('out_inv', 't1w_realign_xfm')]),
```
Comment (oesteban) [inline smriprep/workflows/anatomical.py:620]: I have a vague memory that FreeSurfer's LTAs nomenclature is like the BIDS-Derivatives' `mode-points` of transforms (i.e., the _reference_ is the `to-` field and its data is moved to the _target_ --or `from-`). I'll need to check this offline to make sure we are correctly using and naming these outputs.

Comment (oesteban) [reply smriprep/workflows/anatomical.py:620]: I haven't seen any ApplyTransform based on these new ITK transforms in sMRIPrep, so I'll need to run a full fMRIPrep run with this to make sure we got the direction of the transform correct.

**Comment by oesteban** [comment]
Running this on ds031 right now. I'll report back tomorrow

**Comment by oesteban** [comment]
Okay, I had to switch from ds031 to ds114 as datalad+openneuro wasn't up-to-date when I started. Now I have sMRIPrep run on ds114 - will report soon whether this is working properly.

**Comment by oesteban** [comment]
Because of #192, I haven't locally tested https://github.com/poldracklab/smriprep/pull/188/commits/a5a826326fa3daa1c53e95f7ae59b0e6c28b51b7 - will report back when finished. This should not affect Circle builds as we don't have any multisession / multi-t1 dataset there.

File: smriprep/workflows/anatomical.py (hunk 2 above)
Comment (oesteban) [reply smriprep/workflows/anatomical.py]: Addressed in https://github.com/poldracklab/smriprep/pull/188/commits/8db93dcff385c2aabd15f626e706bcaa8239d7d5. Checked on one session+run of ds031.

**Comment by oesteban** [comment]
@effigies, @mgxd any final comments?

(Mathias, we can release right now and add it later. Otherwise, the changelog will need an update - WDYT?)

**Comment by effigies** [comment]
LGTM.

**Comment by mgxd** [comment]
yeah, this will make it in for 20.1

**Pull request merged**